noxfile.py

```diff
@@ -803,24 +807,59 @@ def notebook(session: nox.Session):
     # when the environment variable is set as it is above,
     # notebooks output a .bytesprocessed and .slotmillis report
     # collect those reports and print a summary
-    _print_performance_report()
+    _print_performance_report("notebooks/")
+
+
+@nox.session(python=SYSTEM_TEST_PYTHON_VERSIONS)
+def benchmark(session: nox.Session):
+    session.install("-e", ".[all]")
+
+    benchmark_script_list = list(Path("scripts/benchmark/").glob("*.py"))
+
+    # Run benchmarks in parallel session.run's, since each benchmark
+    # takes an environment variable for performance logging
+    processes = []
+    for benchmark in benchmark_script_list:
+        process = Process(
+            target=session.run,
+            args=("python", benchmark),
+            kwargs={
+                "env": {
+                    LOGGING_NAME_ENV_VAR: "scripts/benchmark/"
+                    + os.path.basename(benchmark)
+                }
+            },
+        )
+        process.start()
+        processes.append(process)
+
+    for process in processes:
+        process.join()
```

Comment:
Fancy! I think it's smart to run these in parallel, though it might increase the variance a bit due to resource contention.